From: Keith Packard <keithp@keithp.com>
Date: Sun, 4 May 2008 03:23:37 +0000 (-0700)
Subject: Deal with libtool 2.2 which doesn't let us use LT_ variables. (bug 15692)
X-Git-Tag: 2.5.92
X-Git-Url: http://gitweb.freedesktop.org/?p=fontconfig;a=commitdiff;h=19d124dc4f9a68e1c9ddce58bf79a5e11d2ccbfd

Deal with libtool 2.2 which doesn't let us use LT_ variables. (bug 15692)

Libtool-2.2 introduces new restrictions. So now it does not allow LT_*
variables as it includes marcros:

m4_pattern_forbid([^_?LT_[A-Z_]+$])

Rename the LT_ variables to LIBT_ to work around this restriction.
---

--- a/configure.in
+++ b/configure.in
@@ -41,17 +41,17 @@ dnl libtool versioning
 dnl bump revision when fixing bugs
 dnl bump current and age, reset revision to zero when adding APIs
 dnl bump current, leave age, reset revision to zero when changing/removing APIS
-LT_CURRENT=4
-LT_REVISION=0
-AC_SUBST(LT_CURRENT)
-AC_SUBST(LT_REVISION)
-LT_AGE=3
+LIBT_CURRENT=4
+LIBT_REVISION=0
+AC_SUBST(LIBT_CURRENT)
+AC_SUBST(LIBT_REVISION)
+LIBT_AGE=3
 
-LT_VERSION_INFO="$LT_CURRENT:$LT_REVISION:$LT_AGE"
-AC_SUBST(LT_VERSION_INFO)
+LIBT_VERSION_INFO="$LIBT_CURRENT:$LIBT_REVISION:$LIBT_AGE"
+AC_SUBST(LIBT_VERSION_INFO)
 
-LT_CURRENT_MINUS_AGE=`expr $LT_CURRENT - $LT_AGE`
-AC_SUBST(LT_CURRENT_MINUS_AGE)
+LIBT_CURRENT_MINUS_AGE=`expr $LIBT_CURRENT - $LIBT_AGE`
+AC_SUBST(LIBT_CURRENT_MINUS_AGE)
 
 dnl ==========================================================================
 
--- a/fontconfig-zip.in
+++ b/fontconfig-zip.in
@@ -10,7 +10,7 @@ DEVZIP=/tmp/fontconfig-dev-@VERSION@.zip
 cd @prefix@
 rm -f $ZIP
 zip $ZIP -@ <<EOF
-bin/libfontconfig-@LT_CURRENT_MINUS_AGE@.dll
+bin/libfontconfig-@LIBT_CURRENT_MINUS_AGE@.dll
 etc/fonts/fonts.conf
 EOF
 
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -112,7 +112,7 @@ libfontconfig_la_SOURCES = \
 lib_LTLIBRARIES = libfontconfig.la
 
 libfontconfig_la_LDFLAGS =			\
-	-version-info @LT_VERSION_INFO@ -no-undefined $(export_symbols)
+	-version-info @LIBT_VERSION_INFO@ -no-undefined $(export_symbols)
 
 libfontconfig_la_LIBADD = $(ICONV_LIBS) $(FREETYPE_LIBS) $(LIBXML2_LIBS) $(EXPAT_LIBS)
 
@@ -148,6 +148,6 @@ fontconfig.def: $(PUBLIC_FILES) $(PUBLIC
 	(cat $(PUBLIC_FILES) $(PUBLIC_FT_FILES) || echo 'FcERROR ()' ) | \
  	grep '^Fc[^ ]* *(' | sed -e 's/ *(.*$$//' -e 's/^/	/' | \
 	sort; \
-	echo LIBRARY libfontconfig-@LT_CURRENT_MINUS_AGE@.dll; \
-	echo VERSION @LT_CURRENT@.@LT_REVISION@) >$@
+	echo LIBRARY libfontconfig-@LIBT_CURRENT_MINUS_AGE@.dll; \
+	echo VERSION @LIBT_CURRENT@.@LIBT_REVISION@) >$@
 	@ ! grep -q FcERROR $@ || ($(RM) $@; false)
